runtime/rococo/src/lib.rs

```diff
@@ -976,6 +977,7 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
 					RuntimeCall::Slots { .. }
 			),
 			ProxyType::Society => matches!(c, RuntimeCall::Society(..)),
+			ProxyType::OnDemandBidding => matches!(c, RuntimeCall::Scheduler(..)),
```

Comment:
Would be better to limit this to the actual parathread assignment provider. @antonva can we make this more restricted easily? / Does this even work, as allowing calls to the parathread assigner pallet?

Reply:
> I think we should be matching `RuntimeCall::OnDemandProvider`

I didn't see an `OnDemandProvider` exposed at the `runtime/lib.rs` level. And still don't see one.

> not have the `ProxyType` at all in runtimes that don't implement the on demand pallet.

Which ones are these?

Reply:
>I didn't see an OnDemandProvider exposed at the runtime/lib.rs level. And still don't see one.

Ah, sorry, I meant OnDemandAssignmentProvider, which should be defined here: https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/rococo/src/lib.rs#L1449

> Which ones are these?

All but Rococo for now. We need to change the other runtimes when we implement the OnDemandAssignmentProvider on them. Polkadot/Kusama/Westend all just set the parachains assigner as their AssignmentProvider a la: 
https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/polkadot/src/lib.rs#L1380
https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/polkadot/src/lib.rs#L1141